.github/workflows/publish-npm.yaml

```diff
@@ -0,0 +1,60 @@
+permissions: write-all
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+
+jobs:
+
+  publish-to-npm:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Make dist folder
+        run: mkdir dist
+
+      - name: Install wasm-pack
+        uses: jetli/wasm-pack-action@v0.3.0
+        with:
+          version: 'latest'
+      - name: Run wasm-pack for node
+        run: wasm-pack build --target nodejs --release --out-dir dist/node
+        working-directory: prql-js/
+#      - name: Run wasm-pack for web
+#        run: wasm-pack build --target no-modules --release --out-dir dist/web
+#        working-directory: prql-js/
+#      - name: Run wasm-pack for bundler
+#        run: wasm-pack build --target bundler --release --out-dir dist/bundler
+#        working-directory: prql-js/
+
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '18.x'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Publish package on npm
+        run: npm publish
+        working-directory: prql-js/
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Bump npm version
+        run: npm version patch
+        working-directory: prql-js/
+
+      - name: Commit version bump
```

Comment:
How do you see the relationship between the version in the file and the tag of the repo?

Do we tag the repo, and then it bumps the version of the npm package? Is that OK that the tagged repo has the previous version?

(at the moment, this workflow runs on `main` so I guess this would run in a loop, but I'm guessing we remove it from running on `main`?)

FWIW I think this isn't easy given we have a monorepo (though good for other reasons); if we have to do the versions manually for a bit to work it out, then that's OK.

Reply:
Unfortunately NPM's versions ( unlike pythons ) cannot be overwritten, once you publish a 0.2.0 you can never update it, so it's not going to match the PRQL version except for the initial release, we can keep the major and minor the same probably and just bump the patch number.

When committing back using the GITHUB_TOKEN it won't trigger the actions again, so this will publish the package, then increment the number, then commit it back, will that work with the other release scripts we have ?

Reply:
Thanks

> (at the moment, this workflow runs on main so I guess this would run in a loop, but I'm guessing we remove it from running on main?)

Even if it won't retrigger, it will currently create another commit for every commit on main; is this intended?

ref https://github.com/prql/prql/issues/736

Reply:
Well the intent is to increase the patch number on successful publishing of the NPM package in package.json,  the side effect is the extra commit - I'm not sure how to do it otherwise.  This will publish on  every merge re #736

Reply:
Yes, the token is Automation so there is no 2FA required.